### PR TITLE
Add SearchChannels to search for offline channels

### PR DESCRIPTION
**Pull request by saltlands**
Twitch change the GetStreams API to only return live channels. This supplements by allowing to search for offline as well as live channels.

### CODE REVIEW

File: channels.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+// Channel describes a channel from SearchChannel
+type Channel struct {
+	ID           string `json:"id"`
+	GameID       string `json:"game_id"`
+	DisplayName  string `json:"display_name"`
+	Language     string `json:"broadcaster_language"`
+	Title        string `json:"title"`
+	ThumbnailURL string `json:"thumbnail_url"`
+	IsLive       bool   `json:"is_live"`
+	// StartedAt    time.Time `json:"started_at"` // This can be invalid if the channel is not live; omitting it. Prefer GetStreams for live channels.
```
Comment (nicklaw5) [inline channels.go:32]: Can I suggest the following?

```suggestion
	StartedAt    Time `json:"started_at"`
```

This library ships with its own `Time` struct that should handle cases such as these.

See https://github.com/nicklaw5/helix/blob/master/time.go#L21-L37

File: channels.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
+package helix
+
+/*
+This file makes up for a missing API in Helix. This appears to be a recent
+change (as of around 2020-08-03) to the Twitch API, where GetStreams no longer
+returns information about offline channels.
+*/
```
Comment (nicklaw5) [inline channels.go:7]: Can we remove this comment from the source code and instead place it in the description of the PR?

File: channels.go (hunk 1 above)
Comment (saltlands) [reply channels.go:32]: I will change this. FYI, I based this struct on Stream in streams.go, which is using time.Time.

File: channels.go (hunk 2 above)
Comment (saltlands) [reply channels.go:7]: Ah, yes of course. It was for my own notes, and I forgot to remove it.

**Comment by saltlands** [comment]
Rebased from master and applied the suggested fixes. Verified the custom Time type does parse empty dates without error.

File: docs/channels_docs.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Channels Documentation
```
Comment (nicklaw5) [inline docs/channels_docs.md:1]: Can we add a link to this file in the docs readme also? See https://github.com/nicklaw5/helix/blob/master/docs/README.md

Comment (saltlands) [reply docs/channels_docs.md:1]: Done! Thanks for the quick replies :D

**Comment by nicklaw5** [comment]
Thanks :+1:

**Pull request merged**